File: code/language/src/main/java/org/fbme/ide/iec61499/repository/PlatformRepository.java
package org.fbme.ide.iec61499.repository;

import org.fbme.lib.iec61499.DeclarationsScope;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.jetbrains.mps.openapi.model.SModel;
import org.jetbrains.mps.openapi.module.SRepository;

public class PlatformRepository extends PlatformElementsOwner {

    private final SRepository myRepository;

    public PlatformRepository(@NotNull SRepository repository) {
        myRepository = repository;
    }

    public SRepository getMPSRepository() {
        return myRepository;
    }

    @NotNull
    public DeclarationsScope getDeclarationsScope() {
        return new PlatformDeclarationsScope(this, null);
    }

    @NotNull
    public DeclarationsScope getDeclarationScopeFor(@Nullable SModel model) {
        return new PlatformDeclarationsScope(this, model);
    }
}

File: code/language/src/main/kotlin/org/fbme/ide/iec61499/repository/PlatformRepository.kt
package org.fbme.ide.iec61499.repository

import org.fbme.lib.iec61499.DeclarationsScope
import org.jetbrains.mps.openapi.model.SModel
import org.jetbrains.mps.openapi.module.SRepository

class PlatformRepository(val mpsRepository: SRepository) : PlatformElementsOwner() {
    val declarationsScope: DeclarationsScope
        get() = PlatformDeclarationsScope(this, null)

    fun getDeclarationScopeFor(model: SModel?): DeclarationsScope {
        return PlatformDeclarationsScope(this, model)
    }
}
